internal/pkg/provision/check/service.go

```diff
@@ -45,3 +46,65 @@ func ServiceStateAssertion(ctx context.Context, cluster provision.ClusterAccess,
 
 	return nil
 }
+
+// ServiceHealthAssertion checks whether service reached some specified state.
+func ServiceHealthAssertion(ctx context.Context, cluster provision.ClusterAccess, service string, setters ...Option) error {
+	opts := DefaultOptions()
+
+	for _, setter := range setters {
+		setter(opts)
+	}
+
+	cli, err := cluster.Client()
+	if err != nil {
+		return err
+	}
+
+	nodes := make([]string, 0, len(cluster.Info().Nodes))
+
+	for _, node := range cluster.Info().Nodes {
+		if len(opts.Types) > 0 {
+			for _, t := range opts.Types {
+				if node.Type == t {
+					nodes = append(nodes, node.PrivateIP.String())
+				}
+			}
+
+			continue
+		}
+
+		nodes = append(nodes, node.PrivateIP.String())
+	}
+
+	nodesCtx := client.WithNodes(ctx, nodes...)
+
+	servicesInfo, err := cli.ServiceInfo(nodesCtx, service)
+	if err != nil {
+		return err
+	}
+
+	serviceOk := false
+
+	for _, serviceInfo := range servicesInfo {
+		if len(serviceInfo.Service.Events.Events) == 0 {
+			return fmt.Errorf("no events recorded yet for service %q", service)
+		}
+
+		lastEvent := serviceInfo.Service.Events.Events[len(serviceInfo.Service.Events.Events)-1]
+		if lastEvent.State != "Running" {
+			return fmt.Errorf("service %q not in expected state %q: current state [%s] %s", service, "Running", lastEvent.State, lastEvent.Msg)
+		}
+
+		if !serviceInfo.Service.GetHealth().GetHealthy() {
+			return fmt.Errorf("service is not health: %s", service)
+		}
+
+		serviceOk = true
```

Comment:
should we check it's healthy on _all_ the nodes? this seems to succeed if just one node service is healthy?

Reply:
Updated. I think it checks all now?